Disposable Sentry Gun Deployment System

Paxilmaniac opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a new engineer only traitor item, the disposable sentry gun deployment system. It's a red toolbox that works and functions as a normal toolbox, until you attack it on harm intent with a wrench. Upon bashing it with a wrench, the toolbox will transform into a ballistic sentry gun, *theoretically* shooting at your enemies, and not you. The turret only has 100 health however, so it will not last long to anyone with any weapon that can deal enough damage.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Engineer has no role restricted traitor items, which is kinda lame. I think if anything else this will make a good area denial, or getaway tool.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: disposable sentry gun deployment system, a toolbox that transforms into a turret when attacked with a wrench
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'upstream/master'

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/master'

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/master'

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/master'

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/master'

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/master'

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/master'

carda-sometimesknownas-paxilmaniac committed:
need a sentry here

sentry going up

optimumtact commented:
![](https://cdn.discordapp.com/emojis/742440929163477144.gif?v=1)

Commit:
newline moment

oops

RuneQuester commented:
Sentry goin up!

Krysonism commented:
40 hp is too low to be useful.

optimumtact commented:
it's a joke kevin, get a sense of humour

Jared-Fogle commented:
going to tell the crew to stand in front of this and see who it doesnt shoot as a traitor tester

Winterous commented:
This would require an agent id to not be shot by, right?
Also I love it. My only idea for an engineer item was x-ray goggles that fry your eyeballs when hit by emp. Until I realised that the thermal goggles are almost exactly that.

Tharcoonvagh commented:
POOTIS

trexter555 commented:
make it upgradeable with 100 and 500 plasteel for level 2 and 3

RuneQuester commented:
> make it upgradeable with 100 and 500 plasteel for level 2 and 3

YES

Fikou commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -751,6 +751,9 @@
 		return TRUE
 
+/obj/machinery/porta_turret/syndicate/pod/toolbox
+	faction = list(ROLE_TRAITOR)
```
this is pretty bad because its just a traitor tester, you should do something like a var for the person who deployed it which is set on the wrench function and then in initialize do faction |= "[REF(master)]" to make the deployer the only one safe

ATH1909 replied:
and maybe add a way to use the wrench to designate people as being friends?

ATH1909 replied:
you should also make it so that bashing this with a wrench heals it instead of damages it

and since this is an 11 TC item, maybe you should make it drop its toolbox upon destruction?

Fikou commented on code/game/machinery/porta_turret/portable_turret.dm:
```diff
@@ -751,6 +751,9 @@
 		return TRUE
 
+/obj/machinery/porta_turret/syndicate/pod/toolbox
+	faction = list(ROLE_TRAITOR)
```
```suggestion
	var/mob/living/creator

/obj/machinery/porta_turret/syndicate/pod/toolbox/Initialize()
	. = ..()
	if(creator)
		faction |= "[REF(creator)]"
```

Fikou commented on code/game/objects/items/devices/traitordevices.dm:
```diff
@@ -262,3 +262,24 @@ effective or pretty fucking useless.
+		user.visible_message("<span class='danger'>[user] bashes [src] with [I]!</span>", \
+			"<span class='danger'>You bash [src] with [I]!</span>", null, COMBAT_MESSAGE_RANGE)
+		playsound(src, "sound/items/drill_use.ogg", 80, TRUE, -1)
+		new /obj/machinery/porta_turret/syndicate/pod/toolbox(get_turf(src.loc))
```
```suggestion
		var/obj/machinery/porta_turret/syndicate/pod/toolbox/turret
		turret.creator = user
		new turret(get_turf(src.loc))
```

Jared-Fogle replied:
Is this how it works? Wouldn't `turret.creator = user` error since you're trying to set `creator` of null? Wouldn't this make more sense as a parameter to Initialize?

Fikou replied:
yeah probably should be an initialize parameter but im too lazy to update the code suggestion

Fikou replied:
hell you probably dont even need any code on the turret itself and just edit the faction in this proc

Fikou commented:
also it would be cool that if you were the creator of the turret and you wrenched it again after something like a 5 second timer it would return back to tooblxo 😳

ATH1909 commented on code/game/objects/items/devices/traitordevices.dm:
```diff
@@ -262,3 +262,24 @@ effective or pretty fucking useless.
 	update_icon()
+
+/obj/item/storage/toolbox/emergency/turret
+	desc = "You feel the need to hit this one with a wrench"
```
```suggestion
	desc = "You feel a strange urge to hit this with a wrench."
```

ATH1909 commented on code/game/objects/items/devices/traitordevices.dm:
```diff
@@ -262,3 +262,24 @@ effective or pretty fucking useless.
+	new /obj/item/wirecutters(src)
+
+/obj/item/storage/toolbox/emergency/turret/attackby(obj/item/I, mob/living/user, params)
+	..()
```
won't this cause the wrench to be placed inside of the toolbox just before it transforms?

Paxilmaniac replied:
In testing I didn't notice it happening but let me check again

Paxilmaniac replied:
Ah it does, let me fix that

ATH1909 commented:
> make it upgradeable with 100 and 500 plasteel for level 2 and 3

Or, alternatively, port the Gunslinger/mini-sentries instead of the toolbox/normal turrets (wait, do Gunslinger engis still use the red toolbox? I forget). Maybe you could make it resuable(,) then, with a cooldown, a resource cost, and an "only one turret at a time" restriction, because as mentioned before, 11 TC's worth of investment *probably* shouldn't be invested in a single-use item that makes a single structure with only 40 health.

ATH1909 commented:
But yeah, this is an amazing idea, and looking back, I'm quite honestly surprised that nobody had thought of this sooner.

Commit:
Changes the description a bit

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

carda-sometimesknownas-paxilmaniac committed:
buffs turret health and prevents wrench wasting

turret now has 100 health and you no longer put the wrench in the toolbox before it gets turned into a turret

carda-sometimesknownas-paxilmaniac committed:
Merge remote-tracking branch 'origin/buildin-a-sentry' into buildin-a-sentry

Commit:
that last commit didn't change shit in this file

bruh

carda-sometimesknownas-paxilmaniac committed:
turrets target everyone that isn't you

thanks mothblocks

IndieanaJones commented:
https://www.youtube.com/watch?v=2HjLN1TA3bc

Winterous commented:
Ooh, on the "undeploying" idea. A way you could do that pretty cleanly I think is that if you unsecure the turret, ie, turn its power off, then unwrench from the floor, it converts back to the toolbox. This would also allow you to repair minor damage, and because it has such a tiny amount of health, the fact that it resets the damage isn't really cheesable.

Tlaltecuhtli commented:
do syndicate turrets shoot at traitors? maybe give it an item  that  can whitelist others (and urself)

Paxilmaniac commented:
> do syndicate turrets shoot at traitors? maybe give it an item that can whitelist others (and urself)

By default syndie turrets shoot at everyone who doesn't have the syndie faction. This turret shoots at everyone who isn't the spaceman that hit the toolbox with a wrench.

anconfuzedrock commented:
BAP

RuneQuester commented:
Is there a way to move it? Because 11TC for a pretty weak turret that cant be moved is pretty shittty
